release.md

```diff
@@ -0,0 +1,173 @@
+     *Unsupported* (see [the homebrew-simulation issue](https://github.com/osrf/homebrew-simulation/issues/1314) 
+     for more information)
+
+### Metadata for Releasing
+
+All package managers need to define different data fields in order to provide
+the features needed to install and remove packages. In the case of Ignition
+Libraries supported packages:
+
+ * **deb packages**: debian packaging metadata for Debian/Ubuntu packages is
+     stored in a custom way inside the `release` repositories. At the moment of
+     building packages the metadata will be injected in the Jenkins server
+     together with software sources.
+
+     Ignition projects store the Debian/Ubuntu metadata in individual
+     repositories hosted at: https://github.com/ignition-release/ There is a
+     repository for each major version of every Ignition package.
+
+
+ * **brew**: Brew uses custom ruby files to declare software metadata called
+     `Formulae`. The Ignitions libraries Brew Formulae can be found at:
+     https://github.com/osrf/homebrew-simulation/tree/master/Formula
+
+
+### Versioning
+
+A fundamental part of releasing software or binaries is tagging a snapshot of code
+with a given version. There are different types of versions, this section covers
+the ones involved in the Ignition release process.
+
+#### Versions in software code (upstream versions)
+
+The development team of any software product usually declares a version each
+time they want to make explicit for users that a product is somehow ready to be
+used. This version is called `upstream version` (where upstream is the
+development team or authors).
+
+The current upstream version of Ignition libraries can be found in the repository's
+`CMakeLists.txt` file following the CMake `project` declaration:
+
+```cmake
+  project(ignition-fooX VERSION X.Y.Z)
+```
+
+The Ignition version numbers follows the [Semantic
+Versioning](https://semver.org/) scheme so versions are composed of
+`MAJOR.MINOR.PATCH` numbers. The Ignition libraries also support the option of
+creating prereleases. Using [ign-cmake it's
+trivial](https://github.com/ignitionrobotics/ign-cmake/blob/ign-cmake2/examples/prerelease/CMakeLists.txt#L4)
+for the libraries to declare a prerelease number:
+
+```cmake
+ign_configure_project(VERSION_SUFFIX pre1)
+```
+
+#### Versions in binary packages
+
+Binary packages should match exactly the Ignition code version `X.Y.Z`. All
+package systems append a new part to the version called the **release version**
+which is designed to reflect changes in the packaging while keeping the same
+code version `X.Y.Z`.
+
+Debian/Ubuntu packaging use a hyphen to separate code version from the revision
+number (starts with 1) `R`: `X.Y.Z-R`.
+
+Brew bottles revisions are defined in the code of Brew formulas. If there is no
+revision, assume 0. When defined start with 1.
+
+```ruby
+bottle do
+  revision 1
+  sha256 "41af286dc0b172ed2f1ca934fd2278de4a1192302ffa07087cea2682e7d372e3" => :blah
```

Comment:
Would be nice to explain a bit more about what this line is doing

Reply:
b5bbe2e